code/modules/mob/living/living_defense.dm

```diff
@@ -40,6 +40,8 @@
 	//Armor
 	var/damage = P.damage
 	var/flags = P.damage_flags()
+	if(ismech(src))
+		damage = P.damage * P.anti_materiel_potential
```

Comment:
Not a fan of this. Override bullet_act on the mech and put it at the top before calling ..().